docs/tools.md

```diff
@@ -0,0 +1,190 @@
+```
+That's it, each time you'll invoke Lean from within this folder or one of its subfolders,
+it will look for files in the default places and in the folder containing your new `leanpkg.path`
+(the path mentioned in the second line is relative to the folder containing `leanpkg.path`, and `.` means current folder).
+Now check `lean --path` from outside or inside the folder containing `leanpkg.path` to see the difference.
+
+### Keeping compiled versions
+
+Note that `lean` rechecks `test.lean` each time you ask it to check `test2.lean`, 
+even if `test.lean` was not modified since it was last checked. 
+This is clearly a waste of CPU.
+You can ask Lean to remember its work by running `lean --make test.lean`.
+This will create `test.olean` containing all the relevant information from `test.lean` you need in
+`test2.lean`.
+The source file `test.lean` won't be checked again while checking `test2.lean` as long as you don't modify its content.
+
+## Interactive theorem proving
+
+Lean is branded as an *interactive* theorem prover.
+Writing files and asking Lean to check them is not very interactive.
+For instance, you should be able to interactively ask Lean where `max` is defined.
+Remember you typed that `max` name on the first line of `test.lean`, in columns 26 to 28.
+You can run Lean is interactive mode, also known as server mode, using `lean --server`.
+Again not much happens.
+Lean is waiting for instructions or questions on the standard input pipe.
+You can ask it to have a look at `test.lean` by typing:
+```
+{ "command": "sync", "file_name": "test.lean", "seq_num": 1 }
+```
+It will answer a couple of messages, claiming to start working, and then be to done before returning to silence.
+You can then ask for information about what's at column 27 of line 1 of `test.lean` by typing:
+```
+{ "command": "info", "file_name": "test.lean", "line": 1, "column": 27, "seq_num": 2 } 
+```
+Lean's answer will include the location of the file defining `max` as well as the type of
+`max`.
+
+Now tell Lean you want to modify the file `test.lean` to remove the final `m` at the end (you're only pretending to do this modification, nothing will be written in the file here):
+```
+{ "command": "sync", "file_name": "test.lean", "content": "lemma zero_max (m : ℕ) : max 0 m = m :=\nmax_eq_right (nat.zero_le)", "seq_num": 3 }
+```
+Lean will immediately reply with the error message we saw earlier,
+complaining that `nat.zero_le` has type `∀ (n : ℕ), 0 ≤ n` but is expected to have type `0 ≤ m`.
+
+By now you should be tired of interacting directly with Lean's interactive mode.
+It's time to install an editor having some plugin that will do the talking with
+`lean --server`.
+Currently you get to choose between emacs and VS code.
+That editor plugin will also be in charge of finding the `lean` executable and starting `lean --server`
+inside the directory containing the `leanpkg.path` that you carefully crafted in the previous section.
+
+
+##  Handling dependencies
```

Comment:
very minor: there is an extra space here

Reply:
Yes, I was fighting random markdown rendering. Sometimes you need to add random spaces for headers to work. :man_shrugging: